src/routers/alpha-router/functions/best-swap-route.ts

```diff
@@ -451,16 +462,47 @@ export async function getBestSwapRouteBy(
   };
   // If swapping on an L2 that includes a L1 security fee, calculate the fee and include it in the gas adjusted quotes
   if (HAS_L1_FEE.includes(chainId)) {
-    // ensure the gasModel exists and that the swap route is a v3 only route
-    const onlyV3Routes = bestSwap.every(
-      (route) => route.protocol == Protocol.V3
-    );
-    if (gasModel == undefined || !onlyV3Routes) {
+    if (v2GasModel == undefined && v3GasModel == undefined) {
       throw new Error("Can't compute L1 gas fees.");
     } else {
-      gasCostsL1ToL2 = await gasModel.calculateL1GasFees!(
-        bestSwap as V3RouteWithValidQuote[]
-      );
+      const v2Routes = bestSwap.filter((routes) => routes.protocol === Protocol.V2);
+      if (v2Routes.length > 0 && V2_SUPPORTED.includes(chainId)) {
+        if (v2GasModel) {
+          const v2GasCostL1 = await v2GasModel.calculateL1GasFees!(v2Routes as V2RouteWithValidQuote[]);
+          gasCostsL1ToL2.gasUsedL1 = gasCostsL1ToL2.gasUsedL1.add(v2GasCostL1.gasUsedL1);
+          gasCostsL1ToL2.gasUsedL1OnL2 = gasCostsL1ToL2.gasUsedL1OnL2.add(v2GasCostL1.gasUsedL1OnL2);
+          if (gasCostsL1ToL2.gasCostL1USD.currency.equals(v2GasCostL1.gasCostL1USD.currency)) {
+            gasCostsL1ToL2.gasCostL1USD = gasCostsL1ToL2.gasCostL1USD.add(v2GasCostL1.gasCostL1USD);
+          } else {
+            gasCostsL1ToL2.gasCostL1USD = v2GasCostL1.gasCostL1USD;
+          }
+          if (gasCostsL1ToL2.gasCostL1QuoteToken.currency.equals(v2GasCostL1.gasCostL1QuoteToken.currency)) {
+            gasCostsL1ToL2.gasCostL1QuoteToken = gasCostsL1ToL2.gasCostL1QuoteToken.add(v2GasCostL1.gasCostL1QuoteToken);
+          } else {
+            gasCostsL1ToL2.gasCostL1QuoteToken = v2GasCostL1.gasCostL1QuoteToken;
+          }
```

Comment:
Would appreciate some comments on the logic applied here - not sure why we're comparing the currencies for gasCostL1USD.

Reply:
Yep, because it's possible that gasCostL1QuoteToken and gasCostL1QuoteToken have different currencies

Reply:
wdym? those are the same variable